ArtOfIllusion/src/artofillusion/procedural/Procedure.java

```diff
@@ -21,6 +25,77 @@
 
 public class Procedure
 {
+
+  public static TextureParameter[] getTextureParameters(Procedure proc, Object texture)
```

Comment:
Looking better. We're getting there. This method should not be static. Rather: `public TextureParameter[] getTextureParameters()` and internally replace `proc` with `this` (actually, only used on line 31, which can be `Module[] modules = getModules();`)